compiler/test/runnable/test23307.d

```diff
@@ -0,0 +1,93 @@
+// REQUIRED_ARGS: -inline -O
+
+// https://issues.dlang.org/show_bug.cgi?id=23307
+
+version (D_SIMD)
```

Comment:
I've only had a quick skim, but I think you only need to replace this with traits compiles? 
```suggestion
static if (__traits(compiles, int4))
```

Reply:
That's what D_SIMD is for. It's used in many other tests.

Reply:
That version has been appropriately to only mean that the compiler supports the dmd-specific `__simd` extensions. So gdc and ldc won't get to test this otherwise.